crates/rpc-client/src/builtin.rs

```diff
@@ -62,17 +62,15 @@ impl BuiltInConnectionString {
             // reqwest is enabled, hyper is not
             #[cfg(all(not(feature = "hyper"), feature = "reqwest"))]
             Self::Http(url) => {
-                Ok(
-                    alloy_transport::Transport::boxed(
-                        alloy_transport_http::Http::<reqwest::Client>::new(url.clone())
-                    )
-                )
-            },
+                Ok(alloy_transport::Transport::boxed(
+                    alloy_transport_http::Http::<reqwest::Client>::new(url.clone()),
+                ))
+            }
 
             // hyper is enabled, reqwest is not
             #[cfg(feature = "hyper")]
-            Self::Http(_) => Err(TransportErrorKind::custom_str(
-                "hyper not supported by BuiltinConnectionString. Please instantiate a hyper client manually",
+            Self::Http(url) => Ok(alloy_transport::Transport::boxed(
+                alloy_transport_http::HyperTransport::new_hyper(url.clone()),
             )),
```

Comment:
Why was this not enabled? cc @yash-atreya

Reply:
Sorry, I missed that! None of the tests covered BuiltInConnectionString.connect_boxed()

Addded coverage in this PR https://github.com/alloy-rs/alloy/pull/1331